caravel/views.py

```diff
@@ -1441,10 +1451,8 @@ def json_error_response(msg, status=None):
                 'Database with id {} is missing.'.format(database_id),
                 QueryStatus.FAILED)
 
-        if not (self.can_access('all_datasource_access', 'all_datasource_access') or
-                self.can_access('database_access', mydb.perm)):
-            json_error_response(__(
-                "SQL Lab requires the `all_datasource_access` or specific DB permission"))
+        if not self.database_access(mydb):
+            json_error_response(__(DATABASE_ACCESS_ERR.format(mydb.database_name)))
```

Comment:
I think there are 2 layers of babel here... Also babel has its own way of allowing formatted strings as such:
gettext(u'Hello %(name)s!', name='World')

moore about it here:
https://pythonhosted.org/Flask-Babel/